docs/sources/flow/reference/components/loki.source.file.md

```diff
@@ -0,0 +1,96 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/loki.source.file
+title: loki.source.file
+---
+
+# loki.source.file
+
+`loki.source.file` reads log entries from files and forwards them to other
+`loki.*` components.
+
+Multiple `loki.source.file` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.source.file "LABEL" {
+  targets    = TARGET_LIST
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+The component starts a new reader for each of the given `targets` and fans out
+log entries to the list of receivers passed in `forward_to`.
+The list of arguments that can be used to configure the block is presented
+below.
+
+Name         | Type                   | Description          | Default | Required
+------------ | ---------------------- | -------------------- | ------- | --------
+`targets`    | `list(map(string))`    | List of files to read from. | | yes
+`forward_to` | `list(chan loki.Entry)` | List of receivers to send log entries to. | | yes
+
+## Blocks
+
+The `loki.source.file` component doesn't support any inner blocks and is
+configured fully through arguments.
+
+## Exported fields
+
+`loki.source.file` does not export any fields.
+
+## Component health
+
+`loki.source.file` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`loki.source.file` does not expose any component-specific debug information. ????
+
+## Component behavior
+Each element in the list of `targets` as a set of key-value pairs called
+_labels_.
+The set of targets can either be _static_, or dynamically provided periodically
+by a service discovery component such as `discovery.fileglob`. The special
```

Comment:
The fileglob component does not exist yet (but will exist by the next release). Should I remove the example and add it back when it's ready?

Reply:
IMO yes, I'd prefer avoiding referencing unimplemented components in our documentation even if we'll add them before the release (some people may still be reading the `next` version documentation)